sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3361,9 +3361,16 @@ pi_result cuda_piEnqueueMemBufferMap(pi_queue command_queue, pi_mem buffer,
         num_events_in_wait_list, event_wait_list, retEvent);
   } else {
     if (retEvent) {
-      *retEvent =
-          _pi_event::make_native(PI_COMMAND_TYPE_MEM_BUFFER_MAP, command_queue);
-      (*retEvent)->record();
+      try {
+        assert(command_queue);
```

Comment:
Suggest moving it to the begging of the function(~line 3342).

Reply:
Good call. I have moved this assert to the beginning of the function.